app/src/main/java/openfoodfacts/github/scrachx/openfood/models/SendProduct.java

```diff
@@ -208,7 +224,7 @@ public boolean isEqual(SendProduct sp) {
                 .getBrands()) && equalityOfString(this.weight, sp.getWeight()) && equalityOfString(this.weight_unit, sp.getWeight_unit()) &&
                 equalityOfString(this.imgupload_front, sp.getImgupload_front()) && equalityOfString(this.imgupload_nutrition, sp
                 .getImgupload_nutrition
-                ()) && equalityOfString(this.imgupload_ingredients, sp.getImgupload_ingredients()));
+                ()) && equalityOfString(this.imgupload_ingredients, sp.getImgupload_ingredients()) && equalityOfString(this.imgupload_barcode, sp.getImgupload_barcode()));
     }
 
     private boolean equalityOfString(String a, String b) {
```

Comment:
I know it's not part of this PR, but this method **equalityOfString** looks more like a generic utility method. Can you move it as static to Utils.java class, maybe someone else might want to use it later.